packages/react-charting/src/components/SankeyChart/SankeyChart.test.tsx

```diff
@@ -0,0 +1,464 @@
+        name: 'inbox',
+      },
+      {
+        nodeId: 7,
+        name: 'Junk Folder',
+      },
+      {
+        nodeId: 8,
+        name: 'Deleted Folder',
+      },
+      {
+        nodeId: 9,
+        name: 'Clicked',
+      },
+      {
+        nodeId: 10,
+        name: 'Opened',
+      },
+      {
+        nodeId: 11,
+        name: ' No further action  required',
+      },
+    ],
+    links: [
+      {
+        source: 0,
+        target: 6,
+        value: 80,
+      },
+      {
+        source: 1,
+        target: 6,
+        value: 50,
+      },
+      {
+        source: 1,
+        target: 7,
+        value: 28,
+      },
+      {
+        source: 2,
+        target: 7,
+        value: 14,
+      },
+      {
+        source: 3,
+        target: 7,
+        value: 7,
+      },
+      {
+        source: 3,
+        target: 8,
+        value: 20,
+      },
+      {
+        source: 4,
+        target: 7,
+        value: 10,
+      },
+      {
+        source: 5,
+        target: 7,
+        value: 10,
+      },
+
+      {
+        source: 6,
+        target: 9,
+        value: 30,
+      },
+      {
+        source: 6,
+        target: 10,
+        value: 55,
+      },
+      {
+        source: 7,
+        target: 11,
+        value: 60,
+      },
+      {
+        source: 8,
+        target: 11,
+        value: 2,
+      },
+    ],
+  },
+};
+
+describe('Sankey Chart snapShot testing', () => {
+  it('renders Sankey correctly', () => {
+    const component = renderer.create(<SankeyChart data={data} height={500} width={800} />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders Sankey correctly on providing nodecolors and border colors ', () => {
+    const nodeColors = ['#E3008C', '#00A2AD', '#022F22', '#00188F'];
+    const borderColors = ['#002E39', '#43002C', '#3B52B4'];
+
+    const component = renderer.create(
+      <SankeyChart
+        data={dataWithoutColors}
+        height={500}
+        width={800}
+        colorsForNodes={nodeColors}
+        borderColorsForNodes={borderColors}
+      />,
+    );
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+describe('Render calling with respective to props', () => {
+  it('No prop changes', () => {
+    const renderMock = jest.spyOn(SankeyChartBase.prototype, 'render');
+    const props = {
+      data,
+      height: 500,
+      width: 800,
+    };
+    const component = mount(<SankeyChart {...props} />);
+    component.setProps({ ...props });
+    expect(renderMock).toHaveBeenCalledTimes(2);
+    renderMock.mockRestore();
+  });
+
+  it('prop changes', () => {
+    const renderMock = jest.spyOn(SankeyChartBase.prototype, 'render');
+    const props = {
+      data,
+      height: 700,
+      width: 1100,
+    };
+    const component = mount(<SankeyChart {...props} />);
+    component.setProps({ ...props, height: 1000 });
+    expect(renderMock).toHaveBeenCalledTimes(2);
```

Comment:
render is getting called twice for prop change and prop no change. Is this expected.

Reply:
I have updated this as we are not updating the props again, so it will be called only once.